server/src/main/java/io/druid/server/QueryLifecycle.java

```diff
@@ -171,10 +172,11 @@ public void initialize(final Query baseQuery)
     }
 
     this.queryPlus = QueryPlus.wrap(
-        (Query) DirectDruidClient.withDefaultTimeoutAndMaxScatterGatherBytes(
-            baseQuery.withId(queryId),
-            serverConfig
-        )
+        QueryContexts.withMaxQueryTimeout(
```

Comment:
instead, can you rename `DirectDruidClient.withDefaultTimeoutAndMaxScatterGatherBytes(..)` to `DirectDruidClient.withTimeoutAndMaxScatterGatherBytes(..)` and take care of max query timeout also in there? Also, I would move that method from DirectDruidClient to `QueryLifeCycle` , it probably makes more sense to be there.

Reply:
👍